http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriberSpec.scala

```diff
@@ -1828,56 +1831,48 @@ class LeoPubsubMessageSubscriberSpec
   }
 
   it should "handle Azure StartRuntimeMessage and start runtime" in isolatedDbTest {
-    val azurePubSubHandler = makeAzureInterp()
-    val leoSubscriber = makeLeoSubscriber(azureInterp = azurePubSubHandler)
-
+    val azurePubsubHandlerInterp = mock[AzurePubsubHandlerInterp[IO]]
+    val leoSubscriber = makeLeoSubscriber(azureInterp = azurePubsubHandlerInterp)
     val res = for {
       disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
-
       azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
                                                      disk.id,
                                                      azureRegion
       )
-
       runtime <- IO(
         makeCluster(1)
           .copy(status = RuntimeStatus.Starting, cloudContext = CloudContext.Azure(azureCloudContext))
           .saveWithRuntimeConfig(azureRuntimeConfig)
       )
       tr <- traceId.ask[TraceId]
-      _ <- leoSubscriber.messageResponder(StartRuntimeMessage(runtime.id, Some(tr)))
+      _ <- leoSubscriber.messageResponder(StartRuntimeMessage(runtime.id, Some(tr))).attempt
       updatedRuntime <- clusterQuery.getClusterById(runtime.id).transaction
```

Comment:
think we don't need `updatedRuntime <- clusterQuery.getClusterById(runtime.id).transaction` anymore? also what is `attempt` for?

Reply:
So I'm not entirely sure why, but when trying to verify startAndMonitorRuntime using the original runtime as the first argument, the values are different.

The values that change are proxyUrl and scopes. The rest is the same.

It looks ugly, but here's the output in case you were curious...
When line 1851 is: 
`verify(azurePubsubHandlerInterp, times(1)).startAndMonitorRuntime(runtime, azureCloudContext)`


Runtime(1,Some(WorkspaceId(435ea3c2-7d78-41d9-a717-a66623b4d3fb)),RuntimeSamResourceId(067e2867-5d4a-47f3-a53c-fd711529b287),RuntimeName(clustername1),Azure(AzureCloudContext(TenantId(testTenant),SubscriptionId(testSubscription),ManagedResourceGroupName(testMrg))),testServiceAccount@example.com,Some(AsyncRuntimeFields(ProxyHostName(ac043b7b-af40-49f7-a3d3-ab6a9d4e55ab),OperationName(operationName1),stagingbucketname1,Some(IP(numbers.and.dots)))),AuditInfo(user1@example.com,2022-09-30T13:23:34.018010Z,None,2022-09-30T13:23:34.018059Z),None,https://leo/proxy/dsp-leo-test/clustername1/jupyter,Starting,Map(),None,None,List(),None,30,Some(defaultClientId),false,Set(RuntimeImage(Jupyter,init-resources/jupyter-base:latest,Some(/home/jupyter),2022-09-30T13:23:34.018811Z), RuntimeImage(Welder,welder/welder:latest,None,2022-09-30T13:23:34.019269Z)),Set(https://www.googleapis.com/auth/userinfo.email, https://www.googleapis.com/auth/userinfo.profile, https://www.googleapis.com/auth/bigquery, https://www.googleapis.com/auth/source.read_only),false,Map(),RuntimeConfigId(1),false),
[info]     AzureCloudContext(TenantId(testTenant),SubscriptionId(testSubscription),ManagedResourceGroupName(testMrg)),
[info]     cats.mtl.Ask$$anon$2@18eb1428
[info] );
[info] -> at org.broadinstitute.dsde.workbench.leonardo.util.AzurePubsubHandlerInterp.startAndMonitorRuntime(AzurePubsubHandler.scala:181)
[info] Actual invocations have different arguments:
[info] azurePubsubHandlerInterp.startAndMonitorRuntime(
[info]     Runtime(1,Some(WorkspaceId(435ea3c2-7d78-41d9-a717-a66623b4d3fb)),RuntimeSamResourceId(067e2867-5d4a-47f3-a53c-fd711529b287),RuntimeName(clustername1),Azure(AzureCloudContext(TenantId(testTenant),SubscriptionId(testSubscription),ManagedResourceGroupName(testMrg))),testServiceAccount@example.com,Some(AsyncRuntimeFields(ProxyHostName(ac043b7b-af40-49f7-a3d3-ab6a9d4e55ab),OperationName(operationName1),stagingbucketname1,Some(IP(numbers.and.dots)))),AuditInfo(user1@example.com,2022-09-30T13:23:34.018010Z,None,2022-09-30T13:23:34.018059Z),None,https://numbers.and.dots/clustername1,Starting,Map(),None,None,List(),None,30,Some(defaultClientId),false,Set(RuntimeImage(Jupyter,init-resources/jupyter-base:latest,Some(/home/jupyter),2022-09-30T13:23:34.018811Z), RuntimeImage(Welder,welder/welder:latest,None,2022-09-30T13:23:34.019269Z)),Set(https://www.googleapis.com/auth/bigquery, https://www.googleapis.com/auth/source.read_only, https://www.googleapis.com/auth/userinfo.email, https://www.googleapis.com/auth/userinfo.profile),false,Map(),RuntimeConfigId(1),false),
[info]     AzureCloudContext(TenantId(testTenant),SubscriptionId(testSubscription),ManagedResourceGroupName(testMrg)),

Reply:
For the attempt portion, without using attempt, I get an NPE on "fa"

 should handle Azure StartRuntimeMessage and start runtime *** FAILED ***
[info]   java.lang.NullPointerException: Cannot invoke "cats.effect.IO.handleErrorWith(scala.Function1)" because "fa" is null